apps/gitness/src/pages/pull-request/hooks/usePRChecksDecision.ts

```diff
@@ -0,0 +1,174 @@
+import { useEffect, useMemo, useState } from 'react'
+import { TypesPullReq, TypesRepository, useChecksPullReqQuery } from '@harnessio/code-service-client'
+
+import { ExecutionState } from '../../../types'
+import { determineStatusMessage, generateStatusSummary } from '../utils'
+
+interface Check {
+  check: {
+    status: ExecutionState
+  }
+}
+
+interface Data {
+  checks: Check[]
+}
+export function usePRChecksDecision({
+  repoMetadata,
+  pullReqMetadata
+}: {
+  repoMetadata: TypesRepository | undefined
+  pullReqMetadata: TypesPullReq | undefined
+}) {
+  const { data, error, refetch } = useChecksPullReqQuery({
+    repo_ref: `${repoMetadata?.path}/+`,
+    pullreq_number: pullReqMetadata?.number ?? 0,
+    queryParams: { debounce: 500 }
+  })
+  const [count, setCount] = useState(DEFAULT_COUNTS)
+  const [color, setColor] = useState<string>('GREEN_500')
+  const [background, setBackground] = useState<string>('GREEN_50')
+  const [message, setMessage] = useState('')
+  const [complete, setComplete] = useState(true)
+  const [summaryText, setSummaryText] = useState('')
+  const [checkInfo, setCheckInfo] = useState<{
+    title: string
+    content: string
+    color: string
+    status: string
+  }>({ title: '', content: '', color: '', status: '' })
+
+  const status = useMemo(() => {
+    let _status: ExecutionState | undefined
+    const _count = { ...DEFAULT_COUNTS }
+    // @ts-expect-error remove "@ts-expect-error" once CodeServiceClient Response for useChecksPullReqQuery is fixed
```

Comment:
Can you create a ticket under PIPE project and assign it to me with details about exactly what issue we're seeing here mentioning why do we need to add this statement?

Reply:
https://harness.atlassian.net/browse/PIPE-22068